server/spawn-vehicle.lua

```diff
@@ -38,21 +70,105 @@ lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehic
-    if garageType == GarageType.DEPOT and FindPlateOnServer(playerVehicle.props.plate) then -- If depot, check if vehicle is not already spawned on the map
+
+    if garageType == GarageType.DEPOT and FindPlateOnServer(playerVehicle.props.plate) then
         return exports.qbx_core:Notify(source, locale('error.not_impound'), 'error', 5000)
     end
 
     local warpPed = Config.warpInVehicle and GetPlayerPed(source)
-    local netId, veh = qbx.spawnVehicle({ spawnSource = spawnCoords, model = playerVehicle.props.model, props = playerVehicle.props, warp = warpPed})
+    local netId, veh = qbx.spawnVehicle({
+        spawnSource = spawnCoords,
+        model = playerVehicle.props.model,
+        props = playerVehicle.props,
+        warp = warpPed
+    })
+
+    hookPayload.netId = netId
+    hookPayload.veh = veh
+
+    -- Trigger the spawnedVehicle hook
+    triggerEventHooks('garages:spawnedVehicle', hookPayload)
 
-    if Config.doorsLocked then
+    -- Use the doorsLocked value from the payload (which may have been modified by a hook)
+    if hookPayload.doorsLocked then
         SetVehicleDoorsLocked(veh, 2)
     end
 
-    TriggerClientEvent('vehiclekeys:client:SetOwner', source, playerVehicle.props.plate)
+    -- Use the key system from the payload (which may have been modified by a hook)
+    if hookPayload.vehiclekeys then
+        TriggerClientEvent('vehiclekeys:client:SetOwner', source, playerVehicle.props.plate)
+    end
 
+    -- Set additional vehicle states or properties
     Entity(veh).state:set('vehicleid', vehicleId, false)
     setVehicleStateToOut(vehicleId, veh, playerVehicle.modelName)
+
+    -- Trigger an event after the vehicle has been spawned
     TriggerEvent('qbx_garages:server:vehicleSpawned', veh)
+
+    -- Send notification if defined in the hookPayload
+    if hookPayload.Notify.description then
+        exports.qbx_core:Notify(source, hookPayload.Notify.description, hookPayload.Notify.type)
+    end
+
     return netId
-end)
\ No newline at end of file
+end)
+
+-- EXEMPLE
+-- Register the hook with qbx_garages
+exports.qbx_garages:registerHook('garages:spawnVehicle', function(payload)
+    -- Check if the vehicle is being spawned from the 'impound_lspd' garage
+    if payload.garageName == 'impound_lspd' then
+        lib.print.info('Canceling spawn for impound_lspd')
+        payload.cancel = true
+        -- payload.Notify = {description = 'You must request your vehicle from the impound.', type = 'error'}
+        payload.Notify.description = 'You must request your vehicle from the impound.'
+        payload.Notify.type = 'error'
+        return false -- Cancels the spawn
+    end
+
+    return true -- Continue with the spawn
+end)
+
+exports.qbx_garages:registerHook('garages:spawnedVehicle', function(payload)
+    -- Check if the vehicle is being spawned from the 'impound_lspd' garage
+    payload.Notify = {description = 'Enjoy your trip.', type = 'inform'}
+    return true -- Continue with the spawn
+end)
```

Comment:
Examples should be put in the README, not in the code itself.

Reply:
> Examples should be put in the docs, not in the resource itself.

Okay, when I get home from work, I'll do it.